.github/api_best_practices.md

```diff
@@ -0,0 +1,231 @@
+# ACS API Guidelines
+An overview of the set of standardized guidelines and practices to ensure that the APIs across different 
+projects and teams adhere to a common structure and design. This consistency simplifies the process of understanding 
+and maintaining APIs. The guidelines helps identify potential pitfalls and design choices that could lead to issues to 
+help developers avoid common mistakes and build more robust APIs. When in doubt, It is encouraged to seek resolution
+in slack channel [#forum-acs-api-design](https://redhat-internal.slack.com/archives/C05MMG2PP8A).
+
+## Table of Contents
+
+- [Naming Guidelines](#naming-guidelines)
+  - [gRPC Service Name](#grpc-service-name)
+  - [gRPC Method Name](#grpc-method-name)
+  - [gRPC Message Name](#grpc-message-name)
+
+## Naming Guidelines
+
+### gRPC Service Name
+Service name **must** be unique and should use a noun that generally refers to a resource or product component 
+e.g. `DeploymentService`, `ReportService`, `ComplianceService`. Intuitive and well-known short forms or 
+abbreviations may be used in some cases (and could even be preferable) for succinctness 
+e.g. `ReportConfigService`, `RbacService`. 
+All gRPC methods grouped into a single service must generally pertain to the primary resource of the service.
+
+Note:
+- All service defined in versioned package [/proto/api/v2](https://github.com/stackrox/stackrox/blob/master/proto/v2)
+**must not** import from [/proto/storage](https://github.com/stackrox/stackrox/blob/master/proto/storage) and
+[/proto/internalapi](https://github.com/stackrox/stackrox/blob/master/proto/internalapi) packages.
+- All new gRPC methods defined in versioned package [/proto/api/v1](https://github.com/stackrox/stackrox/blob/master/proto/v1) 
+**must not** import from `/proto/storage` and `/proto/internalapi` packages.
+- All services of the type `APIServiceWithCustomRoutes` **must not** import from `/proto/storage` and `/proto/internalapi` packages.
+
+### gRPC Method Name
+Methods should be named such that they provide insights into the functionality. Typically, the method 
+name should follow the _VerbNoun_ convention. For example, `StartComplianceScan`, `RunComplianceScan`, 
+and `GetComplianceScan` are not the same. `StartComplianceScan` must return without waiting for the compliance 
```

Comment:
Do we have a standard verb for a method that starts a task and _does_ wait for it to complete?

Reply:
I think that should be accomplished through the request option. For example, `RunComplianceScan` is ambiguous, however, if its request is:
```
RunComplianceScanRequest {
  bool wait_for_scan_completion
} 
```

then the communication is clear.

Reply:
That looks reasonable to me.  Can that be integrated into this paragraph somehow? e.g. "the method name is ambiguous and should be avoided unless fields in the message help clarify the meaning."